cranelift/interpreter/src/step.rs

```diff
@@ -904,11 +904,20 @@ where
         Opcode::IsInvalid => unimplemented!("IsInvalid"),
         Opcode::Bitcast | Opcode::ScalarToVector => {
             let input_ty = inst_context.type_of(inst_context.args()[0]).unwrap();
-            let arg0 = extractlanes(&arg(0), input_ty)?;
-            let lanes = &arg0
-                .into_iter()
-                .map(|x| DataValue::convert(x, ValueConversionKind::Exact(ctrl_ty.lane_type())))
-                .collect::<ValueResult<SimdVec<DataValue>>>()?;
+            let out_lane_ty = ctrl_ty.lane_type();
+            let nr_out_lane_bytes = out_lane_ty.bytes() as usize;
+            let lanes = &if input_ty.is_vector() {
+                arg(0)
+                    .into_array()?
+                    .chunks(nr_out_lane_bytes)
+                    .map(|bytes| DataValue::read_from_slice_le(&bytes, out_lane_ty))
```

Comment:
Sure, I could certainly do that.

I would, however, like to know where it seems to indicate that the bitcasts on s390x are not handled? It might be useful for me in future since the CI run is indicating green for s390x. It failed on a previous run when I used `DataValue::read_from_slice_ne` variant.

Reply:
I see, thank you for the clarification!

So then is your preference for me to include a panic on big endian bitcasts, or to spend some additional time ensuring that it is correctly handled?

Reply:
Up to you! I'm happy either way.

Reply:
I will give it a whirl. I need to set up qemu to test it and get back to you.

Reply:
I mostly took that test ran it in QEMU s390x and recorded the output. I didn't look at it particularly hard, however I have some faith that the s390x backend would be correct in this regard. If you'd like to look further into this there is a very long [thread](https://github.com/bytecodealliance/wasmtime/issues/4566) where we nailed down the bitcast semantics

Reply:
Thank you, the thread definitely helped me understand the semantics.

I am subsequently able to get the expected output for `bitcast_more_lanes_big`, however, I am not quite as successful with `bitcast_fewer_lanes_big`. If you have the time, could you kindly confirm the expected output for `bitcast_fewer_lanes_big` since I am getting: `Failed test: run: %bitcast_fewer_lanes_big(0x00ff0000000000000000000000000000) == 0x00000000000000ff0000000000000000, actual: 0x000000000000000000ff000000000000`

Reply:
That test does pass on s390x, and at least at a first glance it makes some sense to me, since the first lane glance it appears that that lane gets mirrored in the i64? This at least seems consistent with the behavior described in page 12 of the [Vector Bitcast Presentation](https://github.com/bytecodealliance/wasmtime/issues/4566#issuecomment-1208336548) which essentially makes bitcast a permute? Sorry if I can't help too much here, this stuff really confuses me.

I've tested a few more cases to try and explore this a bit more:
```
; run: %bitcast_fewer_lanes_big(0x00000000000000000000000000000000) == 0x00000000000000000000000000000000
; run: %bitcast_fewer_lanes_big(0x00ff0000000000000000000000000000) == 0x00000000000000ff0000000000000000
; run: %bitcast_fewer_lanes_big(0x00ff00ff000000000000000000000000) == 0x0000000000ff00ff0000000000000000
; run: %bitcast_fewer_lanes_big(0x00ff00ff000000ff0000000000000000) == 0x00ff000000ff00ff0000000000000000
; run: %bitcast_fewer_lanes_big(0x00ff00ff000000ff00000000000000ff) == 0x00ff000000ff00ff00ff000000000000
; run: %bitcast_fewer_lanes_big(0x00ff00ff000000ff00ff0000000000ff) == 0x00ff000000ff00ff00ff0000000000ff
```

Reply:
That is indeed interesting!

I will try to dig a little deeper to see if I can shed some light on this.

Thank you for the additional information, it is really appreciated!

Reply:
Hey,

So I removed the explicit calls to `DataValue::read_from_slice` since it was superfluous. Nevertheless, it seems the output when running the Big endian tests through the interpreter still does not yield the expected results you showed.

I won't hold the fix hostage any longer than I already have, so I have added a panic when attempting to bitcast on vectors when the big endian mem flag is set.

Reply:
I think disabling big endian for now seems like a reasonable solutions. I'm not entirely sure what is up with filetest infrastructure. Thanks for looking into this!

Reply:
No problem at all. Thank you for your patience!